### PR TITLE
MINOR: updated incorrect JavaDocs for joins

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
\cc @miguno @dguy @enothereska

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java [hunk 1]
```diff
@@ -941,7 +941,11 @@ void process(final ProcessorSupplier<? super K, ? super V> processorSupplier,
      * <td></td>
      * </tr>
      * </table>
-     * Both input streams need to be co-partitioned on the join key.
+     * Both input streams (or to be more precise, their underlying source topics) need to have the same number of
+     * partitions.
+     * If this is not the case, you would need to call {@link #through(String)} (for one input stream) before doing the
+     * join, using a pre-created topic with the "correct" number of partitions.
+     * Furthermore, both input streams need to be co-partitioned on the join key.
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java]: could this be a bit clearer? sth like:

> both input streams need to use the same partitioner

**Review by dguy — COMMENTED** [review]
Just a minor comment/question

**Comment by mjsax** [comment]
Updated.

File: streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java [hunk 2]
```diff
@@ -719,19 +708,8 @@ void to(final Serde<K> keySerde,
      * <td></td>
      * </tr>
      * </table>
-     * Both input streams need to be co-partitioned on the join key.
-     * If this requirement is not met, Kafka Streams will automatically repartition the data, i.e., it will create an
-     * internal repartitioning topic in Kafka and write and re-read the data via this topic before the actual join.
-     * The repartitioning topic will be named "${applicationId}-XXX-repartition", where "applicationId" is
```
Comment (enothereska) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java:725]: So we don't want this explanation on repartition topic name anymore?

**Review by dguy — APPROVED** [review]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java (hunk 2 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java:725]: We don't get any repartitioning topics for KTables, thus this explanation does not apply.

It's a KTable invariant that it is partitioned by key -- thus, if you "feed" it into a join, you know that it's partitioned correctly (this was a c&p error for KStream-KStream joins).

**Comment by mjsax** [comment]
@guozhangwang Call for review and merging.

**Review by guozhangwang — APPROVED** [review]
Merged to trunk.

**Comment by mjsax** [comment]
@guozhangwang Can you cherry-pick to `0.10.2`, too?

**Comment by guozhangwang** [comment]
Cherry-picked to 0.10.2.

**Pull request closed without merging**